rust/utils/run-locally/src/cosmos/link.rs

```diff
@@ -0,0 +1,271 @@
+use std::path::Path;
+
+use cosmwasm_schema::cw_serde;
+use hpl_interface::{
+    core,
+    ism::{self},
+};
+
+use super::{cli::OsmosisCLI, crypto::KeyPair, CosmosNetwork};
+
+#[cw_serde]
+pub struct GeneralIsmMessage<T> {
+    pub ism: T,
+}
+
+#[cw_serde]
+pub struct GeneralRouterMessage<T> {
+    pub router: T,
+}
+
+#[cw_serde]
+pub struct GeneralHookMessage<T> {
+    pub hook: T,
+}
+
+#[cw_serde]
+pub struct MockRouterMsg {
+    pub set_route: MockRouterMsgInner,
+}
+
+#[cw_serde]
+pub struct MockRouterMsgInner {
+    pub set: MockDomainRouteSet,
+}
+
+#[cw_serde]
+pub struct MockDomainRouteSet {
+    pub domain: u32,
+    pub route: String,
+}
+
+#[cw_serde]
+pub struct RemoteGasDataConfig {
+    pub remote_domain: u32,
+    pub token_exchange_rate: String,
+    pub gas_price: String,
+}
+
+#[cw_serde]
+pub struct RemoteGasDataConfigExecute {
+    pub set_remote_gas_data_configs: RemoteGasDataConfigExecuteInner,
+}
+
+#[cw_serde]
+pub struct RemoteGasDataConfigExecuteInner {
+    pub configs: Vec<RemoteGasDataConfig>,
+}
+
+#[cw_serde]
+pub struct MockHookQueryMsg {
+    quote_dispatch: MockQuoteDispatch,
+}
+
+#[cw_serde]
+pub struct MockQuoteDispatch {
+    pub metadata: String,
+    pub message: String,
+}
+
+#[cw_serde]
+pub struct GeneralIsmValidatorMessage {
+    pub enroll_validator: EnrollValidatorMsg,
+}
+
+#[cw_serde]
+pub struct EnrollValidatorMsg {
+    pub set: EnrollValidatorSet,
+}
+
+#[cw_serde]
+pub struct EnrollValidatorSet {
+    pub domain: u32,
+    pub validator: String,
+}
+
+fn link_network(
```

Comment:
my understanding is that this function set up and registers a validator, so maybe a better name would be `set_up_validator()`?